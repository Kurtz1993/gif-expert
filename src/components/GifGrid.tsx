import { useFetchGifs } from '../hooks/use-fetch-gifs.hook';
import GifGridItem from './GifGridItem';

interface GifGridProps {
  category: string;
}

const GifGrid = ({ category }: GifGridProps) => {
  const { data, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {loading ? 'Loading...' : data.map(item => <GifGridItem key={item.id} {...item} />)}
      </div>
    </>
  );
};

export default GifGrid;
