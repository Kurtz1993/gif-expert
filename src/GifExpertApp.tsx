import { FunctionComponent, useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

export type GifExpertAppProps = {
  defaultCategories?: string[];
};

const GifExpertApp: FunctionComponent<GifExpertAppProps> = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />

      <AddCategory setCategories={setCategories} />

      <ol>
        {categories.map(category => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
