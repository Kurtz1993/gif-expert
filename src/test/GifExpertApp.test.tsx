import { shallow } from 'enzyme';
import GifGrid from '../components/GifGrid';

import GifExpertApp from '../GifExpertApp';

describe('GifExpertAppComponent', () => {
  test('should render', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should render a list of categories', () => {
    const categories = ['One Punch', 'Katekyo'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper.find(GifGrid).length).toBe(categories.length);
  });
});
