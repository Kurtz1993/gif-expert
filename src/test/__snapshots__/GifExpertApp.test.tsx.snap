// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertAppComponent should render 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
